pkg/kubelet/status_manager.go

```diff
@@ -151,3 +163,14 @@ func (s *statusManager) syncBatch() error {
 	s.DeletePodStatus(podFullName)
 	return fmt.Errorf("error updating status for pod %q: %v", pod.Name, err)
 }
+
+// allTerminated returns true if every status is terminated, or the status list
+// is empty.
+func allTerminated(statuses []api.ContainerStatus) bool {
+	for _, status := range statuses {
+		if status.State.Termination == nil {
```

Comment:
This is only used in the case where deletion timestamp is set, which automatically prevents restarts.  So this method alone doesn't, but in the places where it is called it does